# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/TexUnpackBlob.h
# Commit: 6149f5b15c5e
# Full Hash: 6149f5b15c5e278ef15ea36fa896fa8c146ebdca
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-14 08:32:45
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1640607 - Send SurfaceDescriptors for GPU blitting for video-to-webgl. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101061
# ==============================================================================

diff -r 29fd3a6a7e27 -r 6149f5b15c5e dom/canvas/TexUnpackBlob.h
--- a/dom/canvas/TexUnpackBlob.h	Wed Jan 13 20:14:54 2021 +0000
+++ b/dom/canvas/TexUnpackBlob.h	Wed Jan 13 21:49:40 2021 +0000
@@ -41,6 +41,10 @@
 struct PackingInfo;
 struct DriverUnpackInfo;
 
+Maybe<std::string> BlitPreventReason(int32_t level, const ivec3& offset,
+                                     const webgl::PackingInfo&,
+                                     const TexUnpackBlobDesc&);
+
 class TexUnpackBlob {
  public:
   const TexUnpackBlobDesc& mDesc;